app/experimenter/nimbus-ui/src/components/FormBranches/FormBranch.tsx

```diff
@@ -58,17 +60,23 @@ export const FormBranch = ({
   onFeatureConfigChange: (
     featureConfig: getConfig_nimbusConfig_featureConfig | null,
   ) => void;
+  defaultValues: Record<string, any>;
+  setSubmitErrors: React.Dispatch<React.SetStateAction<Record<string, any>>>;
 }) => {
   const id = fieldNamePrefix;
-
-  const {
-    register,
-    watch,
-    formState: { isSubmitted },
-  } = useFormContext();
-
   const submitErrors = { ...branch.errors };
 
+  const { FormErrors, formControlAttrs, watch } = useCommonNestedForm<
+    typeof branchFieldNames[number]
```

Comment:
Might be handy to separate this out as something like `type BranchFieldName = typeof branchFieldNames[number]` so this part is more concise, but not a biggie

Reply:
Ooo yeah, I like that tweak. I'll make it in the other form files as well while I'm at it for consistency.